packages/input-selection/src/LargeFirstSelection/LargeFirstInputSelector.ts

```diff
@@ -0,0 +1,409 @@
+/* eslint-disable max-params */
+import { Cardano, coalesceValueQuantities } from '@cardano-sdk/core';
+import { ChangeAddressResolver } from '../ChangeAddress';
+import {
+  ImplicitTokens,
+  addTokenMaps,
+  getCoinQuantity,
+  hasNegativeAssetValue,
+  mintToImplicitTokens,
+  sortByAssetQuantity,
+  sortByCoins,
+  stubMaxSizeAddress,
+  subtractTokenMaps,
+  toValues
+} from '../util';
+import {
+  ImplicitValue,
+  InputSelectionParameters,
+  InputSelector,
+  SelectionConstraints,
+  SelectionResult
+} from '../types';
+import { InputSelectionError, InputSelectionFailure } from '../InputSelectionError';
+import { MAX_U64 } from '../RoundRobinRandomImprove';
+import { computeChangeAndAdjustForFee } from '../RoundRobinRandomImprove/change';
```

Comment:
I think it's a little bit odd to import utils from another algorithm. Maybe we could hoist those to `util.ts`

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1621/commits/2e97aa84591000e3c8b25cf0ad999cef94500d60